torch/testing/_internal/common_methods_invocations.py

```diff
@@ -944,6 +944,41 @@ def __init__(self,
                                             **kwargs)
         self.ref = ref
 
+def sample_inputs_foreach(self, device, dtype, N):
+    if dtype in [torch.bfloat16, torch.bool, torch.float16]:
+        tensors = [torch.randn((N, N), device=device).to(dtype) for _ in range(N)]
+    elif dtype in torch.testing.get_all_int_dtypes():
+        tensors = [torch.randint(1, 100, (N, N), device=device, dtype=dtype) for _ in range(N)]
+    else:
+        tensors = [torch.randn((N, N), device=device, dtype=dtype) for _ in range(N)]
+
+    return tensors
```

Comment:
In the future it might be worth testing non-contiguous inputs. But this looks fine for now since it's a refactor of the existing foreach tests

Reply:
Prefer using the test suite's make_tensor() to randint() or randn(), it will handle all dtypes for you. It also has a "discontiguous" kwarg for the future. (That arg will be correctly renamed to noncontiguous in the future, too.)